src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -189,7 +191,7 @@ private DiagnosticInfo ToFileReadDiagnostics(Exception e, CommandLineSourceFile
             return diagnosticInfo;
         }
 
-        internal bool PrintErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput)
+        protected bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput, ErrorLogger errorLogger)
```

Comment:
Seems like a good refactoring, I'll add a new interface `IDiagnosticLogger` with methods to Log Diagnostic and DiagnosticInfo.